### PR TITLE
Fix plugin for IDEA 2018.X. Automate plugin build with gradle.

### PR DESCRIPTION
**Pull request by syakovlevdalet**
Hello,
as we can see plugin no longer works for the new idea versions.
We had fix this issue & also automate plugin build with gradle.

### CODE REVIEW

**Comment by parrt** [comment]
Hi. thanks. @bjansen, can you take a look?

File: src/main/antlr/LexBasic.g4 [hunk 1]
```diff
@@ -131,7 +130,7 @@ fragment CharLiteral	: SQuote ( EscSeq | ~['\r\n\\] )  SQuote	;
 fragment SQuoteLiteral	: SQuote ( EscSeq | ~['\r\n\\] )* SQuote	;
 fragment DQuoteLiteral	: DQuote ( EscSeq | ~["\r\n\\] )* DQuote	;
 fragment USQuoteLiteral	: SQuote ( EscSeq | ~['\r\n\\] )* 			;
-
+DOT:'.';
```
Comment (bjansen) [inline src/main/antlr/LexBasic.g4]: Why did you need to add this exactly?

File: src/main/antlr/STParser.g4 [hunk 2]
```diff
@@ -36,7 +36,7 @@ parser grammar STParser;
 
 options {
 	language=Java;
-	tokenVocab=STLexer;
+//	tokenVocab= STLexer;
```
Comment (bjansen) [inline src/main/antlr/STParser.g4]: I don't think this should be commented out

File: src/main/resources/META-INF/plugin.xml [hunk 3]
```diff
@@ -32,6 +32,7 @@
 
   <project-components>
 	  <component>
+       <!--<implementation-class>org.antlr.jetbrains.st4plugin.STGroupPluginController</implementation-class>-->
```
Comment (bjansen) [inline src/main/resources/META-INF/plugin.xml]: That line is not needed.
```suggestion
```

File: src/main/antlr/STParser.g4 (hunk 2 above)
Comment (bjansen) [reply src/main/antlr/STParser.g4]: I believe you should uncomment the line and add this option to the `generateGrammarSources` target:

```
-lib src/main/antlr
```

File: build.gradle [hunk 4]
```diff
@@ -0,0 +1,58 @@
...
+}
+
+generateGrammarSource {
+    arguments += ["-package", "org.antlr.jetbrains.st4plugin.parsing","-long-messages","-visitor"]
+    outputDirectory = file('src/gen/java/org/antlr/jetbrains/st4plugin/parsing')
+}
+clean {
+    delete 'src/gen/java'
+}
+//apply plugin: 'me.champeau.gradle.antlr4'
```
Comment (bjansen) [inline build.gradle]: That last line is not needed

File: src/main/antlr/LexBasic.g4 (hunk 1 above)
Comment (bjansen) [reply src/main/antlr/LexBasic.g4]: Oh I see, it was to remove a build error. I don't think it should be added here, because it's already defined in `STGLexer.g4`. The correct solution would be to configure the ANTLR4 Gradle plugin to only code for non-included grammars, like this:

```groovy
sourceSets {
    main.java.srcDirs += ['src/gen/java']
    main.antlr.includes = ["ST*.g4"]
}
```

**Comment by bjansen** [comment]
@syakovlevdalet what was it exactly that you changed to make the plugin work with IntelliJ 2018.x?

File: build.gradle (hunk 4 above)
Comment (syakovlevdalet) [reply build.gradle]: agree.

File: src/main/antlr/STParser.g4 (hunk 2 above)
Comment (syakovlevdalet) [reply src/main/antlr/STParser.g4]: agree, I've add this to the `build.gradle`

File: build.gradle (hunk 4 above)
Comment (syakovlevdalet) [reply build.gradle]: agree, removed

**Comment by bjansen** [comment]
Thanks!
